dvc/progress.py

```
@@ -126,6 +126,19 @@ def __exit__(self, typ, value, tbck):
             self.refresh()
         self._lock.release()
 
+    def __call__(self, seq, name="", total=None):
+        if total is None:
+            try:
+                total = len(seq)
+            except TypeError:
+                pass
```

is there a case where it could be useful to silently ignore this?

---

Any iterable, which has no len. Most commonly an iterator.

---

I mean, do you use that anywhere? Doesn't seem so.

---

Not now. However, without this it could be quite a surprise to future user, one may easily decide to    fallback to `progress.update_target()` instead of making `progress.__call__()` work with decorators.

---

If so, it should be tested so there are no surprises for the future user :slightly_smiling_face:

---

Tests don't prevent surprises.

---

Someone is doing:
```python
progress(enumerate(files))
```
Sees an error, falls back to `progress.update_target()`. How tests will help here?

---

You are adding logic that is not used anywhere to prevent surprises for future users, right? That is great, but if you do so, please add a unit test to make sure that it works as expected. Otherwise, it is a dead code that is not used anywhere nor is it tested. Add a test or remove the logic.

---

Ok, removed it. Lesser evil :)